src/js/components/ToggleGroup/ToggleGroup.js

```diff
@@ -0,0 +1,179 @@
+import React, { useCallback, useContext, useState, useRef } from 'react';
+import { ThemeContext } from 'styled-components';
+import { Box } from '../Box';
+import { Keyboard } from '../Keyboard';
+import { ToggleGroupPropTypes } from './propTypes';
+import { StyledButton } from './StyledToggleGroup';
+
+const useControlled = ({ prop, defaultProp, onChange = () => {} }) => {
+  const [uncontrolledProp, setUncontrolledProp] = useState(defaultProp);
+  const controlled = prop !== undefined;
+  const value = controlled ? prop : uncontrolledProp;
+  const handleChange = useCallback(onChange, [onChange]);
+
+  const setValue = useCallback(
+    (nextValue) => {
+      // only update internal value in uncontrolled cases
+      if (!controlled) {
+        setUncontrolledProp(nextValue);
+      }
+      handleChange(nextValue);
```

Comment:
Personally, I'd still recommend we pass this as an object to give use more flexibility in future to pass other things, something like:

```
handleChange({ value: nextValue })
```

Reply:
agree forgot we talked about this!